plant/plants/api/views.py

```diff
@@ -28,6 +44,47 @@ def get_queryset(self):
     def perform_create(self, serializer):
         serializer.save(owner=self.request.user)
 
+    @action(detail=True, methods=["GET"])
+    def plant_stats(self, request, pk=None):
+        start_date = request.query_params.get("start_date")
+        end_date = request.query_params.get("end_date")
+        plant = get_object_or_404(Plant, pk=pk)
```

Comment:
plant = self.get_object() is a convention for retrieving object in a situation like this